ringpop.go

```diff
@@ -213,6 +233,20 @@ func (rp *Ringpop) RegisterListener(l EventListener) {
 	rp.listeners = append(rp.listeners, l)
 }
 
+// getState gets the state of the current Ringpop instance.
+func (rp *Ringpop) getState() state {
+	rp.stateMutex.Lock()
+	defer rp.stateMutex.Unlock()
```

Comment:
By the way, this does not mean that you should never use defer, it gives some nice guarantees. If I remember correctly defer will also be executed if a panic happens. But accessing a single field should not fail catastrophically.